ui/jobs/buff_tracker.ts

```diff
@@ -445,99 +519,106 @@ export class BuffTracker {
       mobLosesEffect: this.mobLosesEffectMap,
     };
 
-    for (let i = 0; i < keys.length; ++i) {
-      const buff = this.buffInfo[keys[i]];
-      buff.name = keys[i];
+    for (const key in this.buffInfo) {
+      const buff = this.buffInfo[key];
+      if (!buff) continue;
+      buff.name = key;
 
-      const overrides = this.options.PerBuffOptions[buff.name] || {};
-      buff.borderColor = overrides.borderColor || buff.borderColor;
-      buff.icon = overrides.icon || buff.icon;
-      buff.side = overrides.side || buff.side || 'right';
-      buff.sortKey = overrides.sortKey || buff.sortKey;
-      buff.hide = overrides.hide === undefined ? buff.hide : overrides.hide;
+      const overrides = this.options.PerBuffOptions[buff.name] ?? null;
+      buff.borderColor = overrides?.borderColor ?? buff.borderColor;
+      buff.icon = overrides?.icon ?? buff.icon;
+      buff.side = overrides?.side ?? buff.side ?? 'right';
+      buff.sortKey = overrides?.sortKey || buff.sortKey;
+      buff.hide = overrides?.hide ?? buff.hide ?? false;
 
       for (const prop in propToMapMap) {
         if (!(prop in buff))
           continue;
-        const key = buff[prop];
+        const key = buff[prop as keyof typeof propToMapMap];
```

Comment:
Yeah, this construction is definitely a little bit weird, where it's using keys to index fields in another type.  It's not a very TypeScript-y way to do things, that's for sure.  I can't come up with a way to not use `as` here, but in general I'm pretty suspicious of using it.

Could you at least move the `as` up to the top, e.g.
```typescript
      for (const propStr in propToMapMap) {
        const prop = propStr as keyof typeof propToMapMap;
```

...that way there's only one `as` and it's a lot less suspicious looking, which makes the code more readable to my eyes.

Can you make `propToMapMap` be marked `as const` as well?